addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java

```diff
@@ -566,110 +656,156 @@ private void readliness(String terminator) throws IOException {
+                    unregisterDeviceStatusListener(deviceStatusListener);
                 }
-                if (!propertiesSet) {
-                    setProperties((H_Message) message);
-                    queueCommand(new SendCommand("Cube(" + getThing().getUID().getId() + ")", new F_Command(),
-                            "Request NTP info"));
+            }
+        }
+    }
+
+    private void processHMessage(H_Message hMessage) {
+        int freeMemorySlotsMsg = hMessage.getFreeMemorySlots();
+        int dutyCycleMsg = hMessage.getDutyCycle();
+        if (freeMemorySlotsMsg != freeMemorySlots || dutyCycleMsg != dutyCycle) {
+            freeMemorySlots = freeMemorySlotsMsg;
+            setDutyCycle(dutyCycleMsg);
+
+            updateCubeState();
+        }
+        if (!propertiesSet) {
+            setProperties(hMessage);
+            queueCommand(
+                    new SendCommand("Cube(" + getThing().getUID().getId() + ")", new F_Command(), "Request NTP info"));
+        }
+    }
+
+    private void processMMessage(M_Message msg) {
+        rooms = new ArrayList<RoomInformation>(msg.rooms);
+
+        if (!roomPropertiesSet) {
+            setProperties(msg);
+        }
+        setProperties(msg);
+        for (DeviceInformation di : msg.devices) {
+            DeviceConfiguration c = null;
+            for (DeviceConfiguration conf : configurations) {
+                if (conf.getSerialNumber().equalsIgnoreCase(di.getSerialNumber())) {
+                    c = conf;
+                    break;
                 }
+            }
 
+            if (c != null) {
+                configurations.remove(c);
             }
-            if (message.getType() == MessageType.M) {
-                M_Message msg = (M_Message) message;
-                rooms = new ArrayList<RoomInformation>(msg.rooms);
 
-                if (!roomPropertiesSet) {
-                    setProperties(msg);
+            c = DeviceConfiguration.create(di);
+            configurations.add(c);
+            c.setRoomId(di.getRoomId());
+            String roomName = "";
+            for (RoomInformation room : msg.rooms) {
+                if (room.getPosition() == di.getRoomId()) {
+                    roomName = room.getName();
                 }
-                setProperties(msg);
-                for (DeviceInformation di : msg.devices) {
-                    DeviceConfiguration c = null;
-                    for (DeviceConfiguration conf : configurations) {
-                        if (conf.getSerialNumber().equalsIgnoreCase(di.getSerialNumber())) {
-                            c = conf;
-                            break;
-                        }
-                    }
+            }
+            c.setRoomName(roomName);
+        }
+    }
 
-                    if (c != null) {
-                        configurations.remove(c);
-                    }
+    private void processNMessage(N_Message nMessage) {
```

Comment:
Better to merge processNMessage &  newInclusionDeviceFound.
For consistency call it processNMessage